### PR TITLE
[Samples][21.core-bot-app-insights] Fix migrated sample

### PR DESCRIPTION
**Pull request by fran893**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/ApplicationInsightsBotTelemetryClient.java [hunk 1]
```diff
@@ -26,17 +28,29 @@
...
+     */
+    public TelemetryConfiguration getTelemetryConfiguration() {
+        return telemetryConfiguration;
+    }
 
     /**
      * Initializes a new instance of the {@link BotTelemetryClient}.
      *
-     * @param withTelemetryClient The telemetry client to forward bot events to.
+     * @param instrumentationKey The instrumentation key provided to create the {@link BotTelemetryClient}.
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/main/java/com/microsoft/bot/applicationinsights/ApplicationInsightsBotTelemetryClient.java:45]:
```suggestion
     * @param instrumentationKey The instrumentation key provided to create the {@link ApplicationInsightsBotTelemetryClient}.

```

File: libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java [hunk 2]
```diff
@@ -180,4 +177,13 @@ public void trackPageViewTest() {
             Assert.assertEquals(0, Double.compare(0.6, pageViewTelemetry.getMetrics().get("metric")));
         }).send(Mockito.any(PageViewTelemetry.class));
     }
+
+    @Test
+    public void flushTest() {
+        botTelemetryClient.flush();
+
+        Mockito.verify(mockTelemetryChannel, invocations -> {
```
Comment (Batta32) [inline libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java:185]: This is just a question, here we are checking against `mockTelemetryChannel` but the one that we need to check if the client right?

Check [JS](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-applicationinsights/tests/telemetryClient.test.js#L103).

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java [hunk 3]
```diff
@@ -194,14 +194,17 @@ public BotTelemetryClient getTelemetryClient() {
...
-            TelemetryConstants.CONVERSATIONNAMEPROPERTY,
-            activity.getConversation().getName()
-        );
-        properties.put(TelemetryConstants.LOCALEPROPERTY, activity.getLocale());
-        properties.put(TelemetryConstants.RECIPIENTIDPROPERTY, activity.getRecipient().getId());
-        properties.put(TelemetryConstants.RECIPIENTNAMEPROPERTY, activity.getRecipient().getName());
+        String fromId = activity.getFrom().getId() != null ? activity.getFrom().getId() : "";
+        properties.put(TelemetryConstants.FROMIDPROPERTY, fromId);
+        String conversationName =
+            activity.getConversation().getName() != null ? activity.getConversation().getName() : "";
```
Comment (Batta32) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java:200]: Is it possible to have all in the same line? Or Checkstyle will complain?

File: samples/21.corebot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/BookingDialog.java [hunk 4]
```diff
@@ -51,9 +51,9 @@ public BookingDialog() {
         
     private CompletableFuture<DialogTurnResult> destinationStep(WaterfallStepContext stepContext) {
         BookingDetails bookingDetails = (BookingDetails) stepContext.getOptions();
-        
-        if (bookingDetails.getDestination() == null) {
-            Activity promptMessage = 
+
```
Comment (Batta32) [inline samples/21.corebot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/BookingDialog.java:54]: Remove empty line

**Review by Batta32 — CHANGES_REQUESTED** [review]
Nit changes, overall LGTM

File: libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java (hunk 2 above)
Comment (fran893) [reply libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java:185]: Here I took as a reference the previous unit tests we migrated which are using the `mockTelemetryChannel`. In JS almost all the unit tests are using the client.

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java (hunk 3 above)
Comment (fran893) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/TelemetryLoggerMiddleware.java:200]: The checkstyle complains 😞

File: libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java (hunk 2 above)
Comment (Batta32) [reply libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java:185]: Ack

**Review by Batta32 — APPROVED** [review]

**Pull request merged**